azkaban-common/src/main/java/azkaban/utils/FileIOUtils.java

```diff
@@ -69,6 +73,32 @@ public static boolean isDirWritable(final File dir) {
     return true;
   }
 
+
+  /**
+   * Return creation time of file.
+   */
+  public static long creationTime(final File file) throws IOException {
+    final BasicFileAttributes attrs = Files
+        .readAttributes(file.toPath(), BasicFileAttributes.class);
+    final FileTime time = attrs.creationTime();
+    return time.toMillis();
+  }
+
+  /**
+   * Return the size of dir in KB.
+   */
+  public static long sizeInKB(final File dir) throws IOException {
```

Comment:
Can we use `FileUtils.sizeOfDirectory` to get the size directly?

Reply:
sizeOfDirectory is not precise to get the read disk usage. E.g directory itself has disk usage but it doesn't count in sizeOfDirectory.

Reply:
high accuracy doesn't hurt, also FileUtils.sizeOfDirectory doesn't really indicate the real disk usage of the directory as size of a file is not equivalent of its associated disk cost. https://superuser.com/questions/94217/why-ls-and-du-show-different-size

Reply:
The downside is unnecessary complexity. 

The referenced article seems to refer to a sparse file. The project files are not sparse files, are they?

Reply:
FileUtils.sizeOfDirectory is a lot slower than running du -sh based on my experiment of reading 10000+ project dir.